### PR TITLE
Apt: Use gpg repo key only for lldpd repository

### PR DESCRIPTION
**Pull request by bastelfreak**
This is based on the apt tips from https://blog.cloudflare.com/dont-use-apt-key/

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -31,14 +31,30 @@
         if ! $gpgkeyfingerprint {
           fail('you must specify a `$gpgkeyfingerprint` when using `$manage_repo` on debian')
         }
+        # place the key in the keyrings directory where apt won't search for keys for all repos
+        file { '/usr/share/keyrings/lldpd.gpg':
+          source => "http://download.opensuse.org/repositories/home:/vbernat/${repourl}/Release.key",
```
Comment (kenyon) [inline manifests/init.pp:36]: https?

File: manifests/init.pp [hunk 2]
```diff
@@ -31,14 +31,30 @@
...
+        }
+        # purge old key files that we installed in previous releases
+        $old_keys = ['/etc/apt/trusted.gpg.d/home_vbernat.gpg', '/etc/apt/trusted.gpg.d/home_vbernat.gpg~',]
+        $old_keys.each |Stdlib::Absolutepath $key| {
+          file { $key:
+            ensure => 'absent',
+          }
+        }
+        # previously managed by apt::key, we need to purge it from the global keyring in /etc/apt/trusted.gpg
+        apt::key { 'EF795E4D26E48F1D7661267B431C37A97C3E114B':
```
Comment (smortex) [inline manifests/init.pp:49]: The CI warnings before the failure makes me think we should `include apt` before this line and the error would go away.

Comment (bastelfreak) [reply manifests/init.pp:50]: fixed it and extended the tests

**Comment by smortex** [comment]
An idea out of nowhere: should we not rather do this once in the apt module so that all modules can benefit from it "transparently" :thinking:  ?

File: manifests/init.pp [hunk 3]
```diff
@@ -31,14 +31,31 @@
...
+            ensure => 'absent',
+          }
+        }
+        # previously managed by apt::key, we need to purge it from the global keyring in /etc/apt/trusted.gpg
+        include apt # required so apt::key can access variables from init.pp
+        apt::key { 'EF795E4D26E48F1D7661267B431C37A97C3E114B':
+          ensure => 'absent',
+        }
         apt::source { 'lldpd':
           location => "http://download.opensuse.org/repositories/home:/vbernat/${repourl}",
```
Comment (kenyon) [inline manifests/init.pp:54]:
```suggestion
          location => "https://download.opensuse.org/repositories/home:/vbernat/${repourl}",
```

File: manifests/init.pp [hunk 4]
```diff
@@ -31,14 +31,31 @@
...
         apt::source { 'lldpd':
-          location => "http://download.opensuse.org/repositories/home:/vbernat/${repourl}",
+          location => "https://download.opensuse.org/repositories/home:/vbernat/${repourl}",
           release  => ' ',
           repos    => '/',
-          key      => {
-            id     => $gpgkeyfingerprint,
-            source => "http://download.opensuse.org/repositories/home:/vbernat/${repourl}/Release.key",
-          },
+          keyring  => '/usr/share/keyrings/lldpd.gpg',
```
Comment (kenyon) [inline manifests/init.pp:57]: The `keyring` parameter was introduced in https://github.com/puppetlabs/puppetlabs-apt/pull/991/commits/0c35168fb0223854fb82d1a61e4769bf2297a17e so I think `metadata.json` also needs to increase the lower bound to v8.1.0.

File: manifests/init.pp [hunk 5]
```diff
@@ -31,14 +31,32 @@
...
+        # place the key in the keyrings directory where apt won't search for keys for all repos
+        # ascii encoded files need to end with *.asc, binary files with .gpg...
+        file { '/usr/share/keyrings/lldpd.asc':
+          source => "https://download.opensuse.org/repositories/home:/vbernat/${repourl}/Release.key",
+          owner  => 'root',
+          group  => 'root',
+          mode   => '0644',
+        }
+        # purge old key files that we installed in previous releases
+        $old_keys = ['/etc/apt/trusted.gpg.d/home_vbernat.gpg', '/etc/apt/trusted.gpg.d/home_vbernat.gpg~',]
```
Comment (alexjfisher) [inline manifests/init.pp:43]: This seems overly verbose.  No need to create a separate variable and iterate.
```suggestion
file { ['/etc/apt/trusted.gpg.d/home_vbernat.gpg', '/etc/apt/trusted.gpg.d/home_vbernat.gpg~']:
  ensure => absent,
}
```

File: manifests/init.pp [hunk 6]
```diff
@@ -31,14 +31,34 @@
...
+        }
+        # purge old key files that we installed in previous releases
+file { ['/etc/apt/trusted.gpg.d/home_vbernat.gpg', '/etc/apt/trusted.gpg.d/home_vbernat.gpg~']:
+  ensure => absent,
+}
+        $old_keys.each |Stdlib::Absolutepath $key| {
+          file { $key:
+            ensure => 'absent',
+          }
+        }
```
Comment (alexjfisher) [inline manifests/init.pp:50]:
```suggestion
        file { ['/etc/apt/trusted.gpg.d/home_vbernat.gpg', '/etc/apt/trusted.gpg.d/home_vbernat.gpg~']:
          ensure => absent,
        }
```

Comment (alexjfisher) [reply manifests/init.pp:50]: Sorry, messed up my `suggestion` last time!

**Review by alexjfisher — APPROVED** [review]

**Pull request merged**